homeassistant/components/sensor/uscis.py

```diff
@@ -0,0 +1,87 @@
+"""
+Support for USCIS Case Status.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.uscis/
+"""
+
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers import config_validation as cv
+from homeassistant.const import CONF_FRIENDLY_NAME
+
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['uscisstatus==0.1.1']
+
+DEFAULT_NAME = "USCIS"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_FRIENDLY_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required('case_id'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and Case Information."""
+    uscis = UscisSensor(config['case_id'], config[CONF_FRIENDLY_NAME])
+    uscis.update()
+    if uscis.valid_case_id:
+        add_devices([uscis])
+    else:
+        _LOGGER.error("Setup USCIS Sensor Fail"
+                      " check if your Case ID is Valid")
+
+
+class UscisSensor(Entity):
+    """USCIS Sensor will check case status on daily basis."""
+
+    HOURS_TO_UPDATE = timedelta(hours=24)
+
+    CURRENT_STATUS = "current_status"
+    LAST_CASE_UPDATE = "last_update_date"
+
+    def __init__(self, case, name):
+        """Initialize the sensor."""
+        self._state = None
+        self._case_id = case
+        self._attributes = None
+        self.valid_case_id = None
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @Throttle(HOURS_TO_UPDATE)
```

Comment:
I did name it like this because timedelta has hour=24so i thought would be better to have it as HOUR_TO_UPDATE instead but if you still think it's better to rename it just let me know @syssi

Reply:
@meauxt We use `MIN_TIME_BETWEEN_UPDATES` everywhere and that can accommodate any time units (hours, minutes, seconds). It may be better, for the sake of consistency, to use the same label.

Reply:
@syssi  @arsaboo  I did update the name of the the variable, Thank you!